frontend/test/pytest/test_transform.py

```diff
@@ -121,44 +164,911 @@ def qfunc(data, x, weights):
+            qml.Barrier(wires=[0, 1])
+            qml.X(0)
+            return qml.expval(qml.Z(0))
+
+        return qfunc
+
+    qnode_control = qnode_builder("default.qubit")
+    qnode_backend = qnode_builder(backend)
+
+    jax_jit = jax.jit(qnode_control)
+    compiled = qjit(qnode_backend)
+
+    expected = jax_jit()
+    observed = compiled()
+    _, expected_shape = jax.tree_util.tree_flatten(expected)
+    _, observed_shape = jax.tree_util.tree_flatten(observed)
+
+    assert np.allclose(expected, observed)
+    assert expected_shape == observed_shape
+
+
+@pytest.mark.xfail(
+    reason="QJIT error ValueError: Passed tape must end in `qml.expval(H)` or qml.var(H)`, where H is of type `qml.Hamiltonian`"
+)
+def test_sign_expand(backend):
+    """Test sign_expand"""
+
+    def qnode_builder(device_name):
+        """Builder"""
+
+        H = qml.Z(0) + 0.5 * qml.Z(2) + qml.Z(1)
+
+        @qml.transforms.sign_expand
+        @qml.qnode(qml.device(device_name, wires=2), interface="jax")
+        def qfunc():
+            qml.Hadamard(wires=0)
+            qml.CNOT(wires=[0, 1])
+            qml.X(2)
+            return qml.expval(H)
+
+        return qfunc
+
+    qnode_control = qnode_builder("default.qubit")
+    qnode_backend = qnode_builder(backend)
+
+    jax_jit = jax.jit(qnode_control)
+    compiled = qjit(qnode_backend)
+
+    expected = jax_jit()
+    observed = compiled()
+    _, expected_shape = jax.tree_util.tree_flatten(expected)
+    _, observed_shape = jax.tree_util.tree_flatten(observed)
+
+    assert np.allclose(expected, observed)
+    assert expected_shape == observed_shape
+
+
+def test_single_qubit_fusion(backend):
+    """Test single_qubit_fusion"""
+
+    def qnode_builder(device_name):
+        """Builder"""
+
+        @qml.transforms.single_qubit_fusion
+        @qml.qnode(qml.device(device_name, wires=1), interface="jax")
+        def qfunc(r1, r2):
+            qml.Hadamard(wires=0)
+            qml.Rot(*r1, wires=0)
+            qml.Rot(*r2, wires=0)
+            qml.RZ(r1[0], wires=0)
+            qml.RZ(r2[0], wires=0)
+            return qml.expval(qml.X(0))
+
+        return qfunc
+
+    r1 = [0.1, 0.2, 0.3]
+    r2 = [0.4, 0.5, 0.6]
+
+    qnode_control = qnode_builder("default.qubit")
+    qnode_backend = qnode_builder(backend)
+
+    jax_jit = jax.jit(qnode_control)
+    compiled = qjit(qnode_backend)
+
+    expected = jax_jit(r1, r2)
+    observed = compiled(r1, r2)
+    _, expected_shape = jax.tree_util.tree_flatten(expected)
+    _, observed_shape = jax.tree_util.tree_flatten(observed)
+
+    assert np.allclose(expected, observed)
+    assert expected_shape == observed_shape
+
+
+def test_split_non_commuting(backend):
+    """Test split non commuting"""
+
+    def qnode_builder(device_name):
+        """Builder"""
+
+        @qml.transforms.split_non_commuting
+        @qml.qnode(qml.device(device_name, wires=6), interface="jax")
+        def qfunc():
+            """Example taken from PL tests"""
```

Comment:
I think this test was written by me. The reason why it is in green is because git sees it being deleted and rewritten instead of being move. We can also remove this comment if you think it is worthwhile.